### DISCUSSION TITLE
Proposal: initialization section on expression-bodied methods

### DISCUSSION THREAD

**Discussion by ctrl-alt-d** [dotnet/csharplang#5922] [Language Ideas]
I miss a "initialization" section on expression-bodied methods. Like this:

```c#
public Order GetOrder(int id)
  with
    using var ctx = DbContextFactory.GetContext();
    // other initializations (consts, vars, ...)
  =>
  ctx
  .First(o => o.Id == id );
```

**Comment by huoyaoyuan** [comment]
Expression bodied members are designed for really simple cases. It's not for such methods.

**Reply by ctrl-alt-d** [reply]
Exactly, with this approach you can also cover medium cases in a predictable pattern: temp vars then final expression.

**Reply by CyrusNajmabadi** [reply]
i guess the point is, the moment you need scoping like this, this is a good time to just use a normal method

**Reply by ctrl-alt-d** [reply]
What would be the main drawback of this syntax?

**Reply by CyrusNajmabadi** [reply]
it's complex and duplicates functionality already available with existing syntax.

**Comment by FaustVX** [comment]
Why not using :
```csharp
public Order GetOrder(int id)
{
    using var ctx = DbContextFactory.GetContext();
    return ctx.First(o => o.Id == id );
}
```

**Reply by ctrl-alt-d** [reply]
Is just about patterns. When you open brackets anything can occur. With proposed syntax you expect the pattern: a section for temp vars and the final expression. More predictable and expressive. In my opinion, of course.
